kratos/includes/gid_gauss_point_container.h

```diff
@@ -370,100 +491,120 @@ class GidGaussPointsContainer
-                                                   ValuesOnIntPoint[index](0,1), ValuesOnIntPoint[index](1,2),
-                                                   ValuesOnIntPoint[index](0,2) );
-                            }
-                            else if(ValuesOnIntPoint[index].size1() ==2
-                                    && ValuesOnIntPoint[index].size2() ==2) {
-                                GiD_fWrite3DMatrix( ResultFile, it->Id(), ValuesOnIntPoint[index](0,0),
-                                                   ValuesOnIntPoint[index](1,1), 0.0,
-                                                   ValuesOnIntPoint[index](0,1), 0.0, 0.0);
-                            }
-                            else if(ValuesOnIntPoint[index].size1() ==1
-                                    && ValuesOnIntPoint[index].size2() ==3) {
-                                GiD_fWrite3DMatrix( ResultFile, it->Id(), ValuesOnIntPoint[index](0,0),
-                                                   ValuesOnIntPoint[index](0,1), 0.0,
-                                                   ValuesOnIntPoint[index](0,2), 0.0, 0.0);
-                            }
-                            else if(ValuesOnIntPoint[index].size1() ==1
-                                    && ValuesOnIntPoint[index].size2() ==4) {
-                                GiD_fWrite3DMatrix( ResultFile, it->Id(), ValuesOnIntPoint[index](0,0),
-                                                   ValuesOnIntPoint[index](0,1), ValuesOnIntPoint[index](0,2),
-                                                   ValuesOnIntPoint[index](0,3), 0.0, 0.0);
-                            }
-                            else if(ValuesOnIntPoint[index].size1() ==1
-                                    && ValuesOnIntPoint[index].size2() ==6) {
-                                GiD_fWrite3DMatrix( ResultFile, it->Id(), ValuesOnIntPoint[index](0,0),
-                                                   ValuesOnIntPoint[index](0,1), ValuesOnIntPoint[index](0,2),
-                                                   ValuesOnIntPoint[index](0,3), ValuesOnIntPoint[index](0,4),
-                                                   ValuesOnIntPoint[index](0,5) );
-                            }
-                        }
-                    }
+                for( auto it = mMeshElements.begin(); it != mMeshElements.end(); it++ ) {
+                    if( !(it->IsDefined(ACTIVE)) || it->Is(ACTIVE) ){
+                        isEntityActive = true;
+                        break;
+                    } 
                 }
-
-                // Resize first matrix to (0,0) for test below
-                ValuesOnIntPoint[0].resize(0, 0, false);
-
             }
             if( mMeshConditions.size() != 0 ) {
                 for( auto it = mMeshConditions.begin(); it != mMeshConditions.end(); it++ ) {
                     if( !(it->IsDefined(ACTIVE)) || it->Is(ACTIVE) ) {
```

Comment:
Once https://github.com/KratosMultiphysics/Kratos/pull/10311 is merged this can be refactored